packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts

```diff
@@ -0,0 +1,214 @@
+import { ZodError } from "zod";
+import {
+  zPluginModuleIDCombo,
+  zPluginType,
+  zPluginPublisher,
+  zPluginInstallOptions,
+  zPluginPermission,
+  zPluginManifestCliOption,
+  zPluginManifestCliCommand,
+  zPluginPackageTCore,
+  zPluginConfigField,
+  zPluginStorageItemSet,
+  zPluginModule,
+  zPlugin,
+  zModuleMessageOptions,
+  zPluginModuleListItem,
+  zPluginModuleList,
+} from "./Plugin.types";
+
+describe("zPluginModuleIDCombo", () => {
+  test("invalid id", () => {
+    const data = "123:aaa";
+    try {
+      zPluginModuleIDCombo.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe("Invalid Plugin ID");
+    }
+  });
+});
+
+describe("zPluginType", () => {
+  test("invalid option", () => {
+    const data = "a";
+    try {
+      zPluginType.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe(
+        "Invalid enum value. Expected 'action-trigger' | 'action-type' | 'database' | 'decoder' | 'encoder' | 'filesystem' | 'hook' | 'navbar-button' | 'page' | 'service' | 'sidebar-button' | 'sidebar-footer-button'"
```

Comment:
Don't add all the values in test, it's going to be difficult to change it later on